functests/cluster_upgrade_test.go

```diff
@@ -0,0 +1,76 @@
+package functests_test
+
+import (
+	"flag"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	tests "github.com/openshift/ocs-operator/functests"
+
+	deploymanager "github.com/openshift/ocs-operator/pkg/deploy-manager"
+)
+
+var _ = Describe("Cluster upgrade", func() {
+	flag.Parse()
+
+	BeforeEach(func() {
+		RegisterFailHandler(Fail)
+	})
+
+	Describe("ocs", func() {
+
+		BeforeEach(func() {
+			By("Preparing for upgrade. Uninstall the current cluster")
+			tests.AfterTestSuiteCleanup()
+			By("Reinstall a fresh cluster based on upgrade_from image")
+			tests.BeforeUpgradeTestSuiteSetup()
+		})
+
+		Context("upgrade cluster", func() {
+			It("and verify deployment status", func() {
+				if tests.UpgradeFromOcsRegistryImage == "" || tests.UpgradeFromLocalStorageRegistryImage == "" {
+					Skip("Condition not met for upgrade")
+				}
+				deployManager, err := deploymanager.NewDeployManager()
+				Expect(err).To(BeNil())
+
+				By("Getting the ocs-catalogsource pod image before the upgrade")
+				ocsImage, err := deployManager.GetDeploymentImage("ocs-operator")
+
+				By("Getting the current csv before the upgrade")
+				csv, err := deployManager.GetCsv()
+				Expect(err).To(BeNil())
+				
+				By("Upgrading OCS with OLM to the current version from upgrade_from version")
+				err = deployManager.UpgradeOCSWithOLM(tests.OcsRegistryImage, tests.LocalStorageRegistryImage, tests.OcsSubscriptionChannel)
+				Expect(err).To(BeNil())
+
+				By("Waiting for OCS CSV to be posted and installed")
+				err = deployManager.WaitForCsvUpgrade(csv.Name, tests.OcsSubscriptionChannel)
+				Expect(err).To(BeNil())
+
+				By("Waiting for ocs-operator, rook-ceph-operator and noobaa-operator to come online.")
+				err = deployManager.WaitForOCSOperator()
+				Expect(err).To(BeNil())
+
+				By("Verifying ocs-operator deployment has been upgraded to the new image")
+				upgradedOcsImage, err := deployManager.GetDeploymentImage("ocs-operator")
+				Expect(err).To(BeNil())
+				Expect(upgradedOcsImage).ToNot(Equal(ocsImage))
```

Comment:
do we need this check? isn't this handled by the `VerifyComponentOperators` function? 

also, checking "!= old image" could lead to issues if both the old and new CSV happen to have the same ocs-operator container image.  that's techincally allowed and possible. we could have a release that only iterates on the rook or noobaa operator for example without updating the ocs-operator image at all. It would still be a new CSV version though.

Reply:
Ok, that makes sense. Removed the image check. VerifyComponentOperators is enough.